docs/Contributing.adoc

```diff
@@ -6,11 +6,38 @@ Herein lies the contribution guidelines for helping out with this project. Do ta
 
 === General rules
 
-* The Ansible <<coc,Code of Conduct>> still applies.
-* For git messages, branch names, ..., follow <<gitstyle,Git Style Guide>>
-* Should you wish to work on a completely new standard, GREAT, but please file an issue for tracking.
+* The Ansible link:https://docs.ansible.com/ansible/latest/community/code_of_conduct.html[Code of Conduct] still applies.
+* For git messages, branch names, ..., follow link:https://github.com/redhat-cop/agnosticd/blob/development/docs/git-style-guide.adoc[Git Style Guide].
+* Should you wish to work on a completely new standard, GREAT, but please file an **issue** for tracking. In your Pull Request, specify `closes #NUM` to automatically close the issue when the PR is merged.
 * To contribute, fork and make a pull request against the `development` branch.
-* All tasks should be in YAML literal.
+*  Use link:https://asciidoctor.org/docs/asciidoc-writers-guide/[asciidoc] for documentation.
+* Pull requests should only change one role, one config, or one role as applied to one config. There may be exceptions to this rule, but you must explain why a special exception should apply to a PR. Ask yourself, « Can this pull request be separeted in several smaller pull requests ? »
+* Pull Request titles:
+** Must start with `Add | Change | Fix | HOTFIX | Remove`
+** followed by `config | workload | role | core | test | documentation`
+** followed by the **name** of the config, workload, cloud provider, test or documentation
+** followed by a short description
+* Pull Request must be tested. Explain how you tested your change.
+* Owner of any new role or config must be identified.
+* If your Pull Request is not ready for review, open it as link:https://github.blog/2019-02-14-introducing-draft-pull-requests/[Draft] or add `WIP` in the title.
+
+=== Code Quality Rules
+
+. a YAML `.yamllint` file should be added to every role and config when any substantial change is applied to the role or config, all new roles and configs must include a `.yamllint`.
```

Comment:
We want it globally, but we have to get there step by step. This is meant to start the process with an aim to later move the `.yamllint` to something more global.

Reply:
Yes I echo @wkulhanek comment - surely we can have one DRY `.yamllint`

Reply:
Perhaps offer Tok's global `.yamllint` that's only (for now) applied via a symlink?

Reply:
Today we cannot have one .yamllint for all the configs and roles, since most of them are not valid according to yamllint.

@jkupferer 's change in the static tests to apply the tests per config or per role *if* there is a `.yamllint` is a way to move forward with this and have hopefully in the long term one `.yamllint` to rule them all.

That change is already in place and allow us at least for new stuff, to check the YAML style and syntax.